lib/ld-eventsource/client.rb

```diff
@@ -163,12 +183,56 @@ def on_error(&action)
     #
     def close
       if @stopped.make_true
-        @cxn.close if !@cxn.nil?
-        @cxn = nil
+        reset_http
       end
     end
 
     private
+    
+    def reset_http
+      @http_client.close if !@http_client.nil?
+      @cxn = nil
+      @buffer = ""
+    end
+    
+    def read_lines
+      Enumerator.new do |gen|
+        loop do
+          line = read_line
+          break if line.nil?
+          gen.yield line
+        end
+      end
+    end
+    
+    def read_line
+      loop do
+        @lock.synchronize do
+          i = @buffer.index(/[\r\n]/)
+          if !i.nil? && !(i == @buffer.length - 1 && @buffer[i] == "\r")
+            i += 1 if (@buffer[i] == "\r" && @buffer[i + 1] == "\n")
+            return @buffer.slice!(0, i + 1).force_encoding(Encoding::UTF_8)
```

Comment:
This may have been another pre-existing problem in my code, but I don't think we want to retain the linefeeds in the returned string. `EventParser` doesn't do anything to explicitly strip them out, and even though it's matching the line to a regex that ends in `$`, the regex engine does _not_ consider `\r` to be a line ending (at least if you're running on Linux) and so when it parses `data: foo\r\n` the field value will end up being `foo\r`. If the old code did have this problem, I think the only reason we didn't get bitten by it is that our Streamer doesn't send CR's.

So we could either explicitly define this method to say it includes the line endings, and update EventParser to strip them, or define this method to say it does not include the line endings and modify this logic accordingly.

Reply:
https://github.com/launchdarkly/ruby-eventsource/blob/05d8e5b31c44a151dc8f76e4c186f53551e30596/lib/ld-eventsource/impl/event_parser.rb#L34

I'm confused does this not strip it out? "it will remove \n, \r, and \r\n"

Reply:
Ah, sorry, yes it will. I just completely missed that line.